gradoop-examples/src/main/java/org/gradoop/benchmark/sampling/SamplingBenchmark.java

```diff
@@ -0,0 +1,421 @@
+    /** RandomVertexNeighborhoodSampling enum constant */
+    RANDOM_VERTEX_NEIGHBORHOOD_SAMPLING(RandomVertexNeighborhoodSampling.class.getSimpleName()),
+    /** RandomVertexSampling enum constant */
+    RANDOM_VERTEX_SAMPLING(RandomVertexSampling.class.getSimpleName());
+
+    /** Property denoting the simple classname of a sampling algorithm */
+    private final String name;
+
+    /**
+     * Enum constructor
+     * @param name simplified class name
+     */
+    Algorithm(String name) {
+      this.name = name;
+    }
+  }
+
+  static {
+    OPTIONS.addOption(Option.builder(OPTION_INPUT_PATH)
+      .longOpt("input")
+      .hasArg()
+      .desc("Path to directory containing csv files to be processed")
+      .required()
+      .build());
+    OPTIONS.addOption(Option.builder(OPTION_SELECTED_ALGORITHM)
+      .longOpt("algorithm")
+      .hasArg()
+      .desc("Positive integer selecting a sampling algorithm")
+      .required()
+      .build());
+    OPTIONS.addOption(Option.builder(OPTION_CONSTRUCTOR_PARAMS)
+      .longOpt("params")
+      .hasArgs()
+      .desc("Whitespace separated list of algorithm parameters")
+      .required()
+      .build());
+    OPTIONS.addOption(OPTION_OUTPUT_PATH, "output", true,
+      "Path to directory where resulting graph sample, benchmark file and graph " +
+        "statistics are written to. (Defaults to " + OUTPUT_PATH_DEFAULT + ")");
+    OPTIONS.addOption(OPTION_COMPUTE_STATISTICS, false,
+      "If provided, graph statistics for input and sample graph are computed.");
+  }
+
+  /**
+   * Main program to run the benchmark. Required arguments are a path to CSVDataSource compatible
+   * files that define a graph, an integer defining the sampling algorithm to be tested and a list
+   * of parameters for the constructor of the sampling class.
+   * Other arguments are the available options.
+   *
+   * @param args program arguments
+   */
+  public static void main(String[] args) throws Exception {
+    CommandLine cmd = parseArguments(args, SamplingBenchmark.class.getName());
+
+    if (cmd == null) {
+      System.exit(1);
+    }
+
+    readCMDArguments(cmd);
+
+    // set flink environment
+    ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+    GradoopFlinkConfig conf = GradoopFlinkConfig.createConfig(env);
+
+    // read logical graph
+    DataSource source = new CSVDataSource(INPUT_PATH, conf);
+    LogicalGraph graph = source.getLogicalGraph();
+
+    // instantiate selected sampling algorithm and create sample
+    SamplingAlgorithm algorithm = instantiateSelectedSamplingAlgorithm();
+    LogicalGraph graphSample = algorithm.execute(graph);
+
+    // write data to sink, overwrite if necessary
+    DataSink sink = new CSVDataSink(OUTPUT_PATH + OUTPUT_PATH_GRAPH_SAMPLE_SUFFIX, conf);
+    sink.write(graphSample, true);
+
+    env.execute();
+
+    writeBenchmark(env, algorithm);
+
+    if (COMPUTE_STATISTICS) {
```

Comment:
This isn't necessary here. `StatisticsRunner` can be executed by its own within a second call.

Reply:
I agree. I added the statistics runner because it was requested by @matthiastaeschner .

Reply:
apparently there was a misunderstanding here. i am removing the statistics runner.